internal/config/config.go

```diff
@@ -98,6 +99,10 @@ func Default(path string) (*viper.Viper, string) {
 }
 
 func New(path string) (*Config, error) {
+	if strings.HasPrefix(path, "~") {
+		return nil, fmt.Errorf("%w: config path: %s", errTildeUsage, path)
+	}
```

Comment:
Why forbidding it?

I would have used something like that.

```suggestion
	path, tildeFound := strings.CutPrefix(path, "~")
	if tildeFound {
		path = os.UserConfigDir() + "/" + path
	}
```

https://pkg.go.dev/os#UserConfigDir

Any reason?

Also, you could use filepath, no? Either:
- https://pkg.go.dev/path/filepath#Clean
- https://pkg.go.dev/path/filepath#Abs

It's worth trying

Or, maybe, I'm totally wrong and I'm missing something.

Reply:
My reasoning is that `~`/`$HOME` expansion is the responsibility of the shell IMHO, and I don't want to have to handle it in my code. It's less logic to consider, less possibility for failures, ... :)

Now that I think about it, I think that the cache path from the configuration won't work if it has `$HOME` either, because I'm not expanding it in this code 🤔 But that would probably be better to support `$HOME` still.

Reply:
I think I'll create a new PR later to support `$HOME` expansion, based on https://pkg.go.dev/os#UserHomeDir and https://pkg.go.dev/os#ExpandEnv.

That feels like an acceptable compromise to me

Reply:
It does seem like `~` should be as supported as `$HOME` tho, they are both defined in the POSIX: https://pubs.opengroup.org/onlinepubs/9799919799/

Buut I don't want to do string manipulation, and since `~` is equivalent enough to `$HOME`, I'll just forbid one in favor of the other. ⚖️

Reply:
Thanks for making me consider that more seriously @ccoVeille 🌷 📖

Reply:
cc https://github.com/nobe4/gh-not/pull/250

Reply:
> But I don't want to do string manipulation, and since ~ is equivalent enough to $HOME, I'll just forbid one in favor of the other. ⚖️

Please adapt the tilde error message to suggest using $HOME

Reply:
Already in https://github.com/nobe4/gh-not/pull/247/files#diff-dd7b39f2e0f33d3ad65c6be4bbc45b531d870ea2811bc82768d2ff0c57b463eaR20